zebra-network/src/peer_set/inventory_registry/tests/vectors.rs

```diff
@@ -0,0 +1,198 @@
+    let test_change = InventoryStatus::new_advertised(test_hash, test_peer);
+
+    let (mut inv_registry, inv_stream_tx) = new_inv_registry();
+
+    let receiver_count = inv_stream_tx
+        .send(test_change)
+        .expect("unexpected failed inventory status send");
+    assert_eq!(receiver_count, 1);
+
+    inv_registry
+        .update()
+        .await
+        .expect("unexpected dropped registry sender channel");
+
+    assert_eq!(
+        inv_registry.advertising_peers(test_hash).next(),
+        Some(&test_peer),
+    );
+    assert_eq!(inv_registry.missing_peers(test_hash).count(), 0);
+}
+
+/// Check inventory registration for one missing hash/peer.
+#[tokio::test]
+async fn inv_registry_one_missing_ok() {
+    let test_hash = InventoryHash::Block(block::Hash([0; 32]));
+    let test_peer = "1.1.1.1:1"
+        .parse()
+        .expect("unexpected invalid peer address");
+    let test_change = InventoryStatus::new_missing(test_hash, test_peer);
+
+    let (mut inv_registry, inv_stream_tx) = new_inv_registry();
+
+    let receiver_count = inv_stream_tx
+        .send(test_change)
+        .expect("unexpected failed inventory status send");
+    assert_eq!(receiver_count, 1);
+
+    inv_registry
+        .update()
+        .await
+        .expect("unexpected dropped registry sender channel");
+
+    assert_eq!(inv_registry.advertising_peers(test_hash).count(), 0);
+    assert_eq!(
+        inv_registry.missing_peers(test_hash).next(),
+        Some(&test_peer),
+    );
+}
+
+/// Check inventory registration for one hash/peer prefers missing over advertised.
+#[tokio::test]
+async fn inv_registry_prefer_missing_ok() {
+    inv_registry_prefer_missing_order(true).await;
+    inv_registry_prefer_missing_order(false).await;
+}
+
+async fn inv_registry_prefer_missing_order(missing_first: bool) {
+    let test_hash = InventoryHash::Block(block::Hash([0; 32]));
+    let test_peer = "1.1.1.1:1"
+        .parse()
+        .expect("unexpected invalid peer address");
+
+    let missing_change = InventoryStatus::new_missing(test_hash, test_peer);
+    let advertised_change = InventoryStatus::new_advertised(test_hash, test_peer);
+
+    let (mut inv_registry, inv_stream_tx) = new_inv_registry();
+
+    let changes = if missing_first {
+        [missing_change, advertised_change]
+    } else {
+        [advertised_change, missing_change]
+    };
+
+    for change in changes {
+        let receiver_count = inv_stream_tx
+            .send(change)
+            .expect("unexpected failed inventory status send");
+        assert_eq!(receiver_count, 1);
+    }
+
+    // TODO: also test with updates after each change
+    inv_registry
+        .update()
+        .await
+        .expect("unexpected dropped registry sender channel");
+
+    assert_eq!(inv_registry.advertising_peers(test_hash).count(), 0);
+    assert_eq!(
+        inv_registry.missing_peers(test_hash).next(),
+        Some(&test_peer),
+    );
```

Comment:
Optional: same as above:

```suggestion
    assert_eq!(
        inv_registry.missing_peers(test_hash).collect::<Vec<_>>(),
        vec![&test_peer],
    );
```